gh-95273: Normalise sqlite3 reference wording (#95274)
python/cpython: Doc/library/sqlite3.rst

.. data:: sqlite_version

>    Version number of the run-time SQLite library as a :class:`string <str>`.

Reviewer: ```suggestion
   Version number of the runtime SQLite library as a :class:`string <str>`.
```

.. data:: sqlite_version_info

>    Version number of the run-time SQLite library as a :class:`tuple` of

Reviewer: ```suggestion
   Version number of the runtime SQLite library as a :class:`tuple` of
```

   .. attribute:: row_factory

      A callable that accepts two arguments,
      a :class:`Cursor` object and the raw row results as a :class:`tuple`,

>       and return any object that fit the application programmers need.

Reviewer 2: I'm rather confused by what

> return any object that fit the application programmers need.

is supposed to mean; aside from the grammar issues, it is unclear to me what I as a user am supposed to do with this.

Author: Well. I also find it hard to word this. The row factory can return whatever the programmer wishes. For one application, returning a string may be the right thing. For another application, returning a dataclass, or a complex object, or just a named tuple may be right.

Author: Perhaps just: _"return a row object tailored to the application specific need"_

Reviewer: `and returns an arbitrary object representing a SQLite row`?

Author: Hm, "arbitrary" sounds ... arbitrary :) How about "returns a custom object", or "application specific object"?

Author: BTW, the "representing an SQLite row" suffix is definitely an improvement. 🐿️

Reviewer: > Hm, "arbitrary" sounds ... arbitrary :) How about "returns a custom object", or "application specific object"?

"Custom object representing an SQLite row" works well imo, or you could just go with "object representing an SQLite row"

> 🐿️

Dammit, will I ever remember 😄

Author: Great, I'll land this using the former variant. Thank you so much, both of you :)

   .. attribute:: text_factory

>       A callable that accept a :class:`bytes` parameter and return a text

Reviewer 2: ```suggestion
      A callable that accepts a :class:`bytes` parameter and returns a text
```

Grammar (singular/plural agreement)

      representation of it.
      The callable is invoked for SQLite values with the ``TEXT`` data type.
      By default, this attribute is set to :class:`str`.
      If you want to return ``bytes`` instead, set *text_factory* to ``bytes``.

      Example:

      .. literalinclude:: ../includes/sqlite3/text_factory.py

   .. attribute:: total_changes

      Return the total number of database rows that have been modified, inserted, or
      deleted since the database connection was opened.

   .. method:: iterdump

      Return an :term:`iterator` to dump the database as SQL source code.
      Useful when saving an in-memory database for later restoration.

>       Similar to the :kbd:`.dump` command in the :program:`sqlite3` shell.

Author: The original wording is wrong; `iterdump` does not handle all corner-cases, so it's better to say "similar" than "the same".

>       Similar to the :kbd:`.dump` command in the :program:`sqlite3` shell.

Reviewer 2: ``.dump`` is not, as far as I'm aware, a keyboard shortcut. If its a GUI element, then it should be

```suggestion
      Similar to the :guilabel:`.dump` command in the :program:`sqlite3` shell.
```

or if its a CLI command, then just the regular

```suggestion
      Similar to the ``.dump`` command in the :program:`sqlite3` shell.
```

Author: It's a CLI command, so let's just use double backticks.

   .. method:: fetchone()

>       Fetch the next row of a query result set as a :class:`tuple`.

Author: We _could_ also mention that the returned row may be produced by a custom `row_factory`.

Author: Ditto for fetchmany and fetchall.

      Return :const:`None` when no more data is available.

   .. method:: fetchmany(size=cursor.arraysize)

      Fetch the next set of rows of a query result as a :class:`list`.

>       Return an empty list when no more rows are available.

Reviewer: ```suggestion
      Return an empty list if no more rows are available.
```

>       Return an empty list is when no more rows are available.

Reviewer 2: ```suggestion
      Return an empty list when no more rows are available.
```

Fix typo

      The number of rows to fetch per call is specified by the *size* parameter.
      If *size* is not given, :attr:`arraysize` determines the number of rows
      to be fetched.

>       If fewer than *size* rows are available, fewer rows are returned.

Reviewer: I'm not sure if this is accurate, but maybe:

```suggestion
      If fewer than *size* rows are available, as many rows as are available are returned.
```

   .. method:: fetchone()

      Fetch the next row of a query result set as a :class:`tuple`.

>       Return :const:`None` when no more data is available.

Reviewer: Oops I missed one

```suggestion
      Return :const:`None` if no more data is available.
```

   .. method:: fetchall()

      Fetch all (remaining) rows of a query result as a :class:`list`.

>       Return an empty list when no rows are available.

Reviewer: ```suggestion
      Return an empty list if no rows are available.
```
